lib/safe-pg-migrations/plugins/statement_insurer.rb

```diff
@@ -84,6 +84,30 @@ def add_index(table_name, column_name, **options)
       without_timeout { super(table_name, **options) }
     end
 
+    ruby2_keywords def rename_table(table_name, new_name, skip_view: false)
+      # raise 'DDL transaction is disabled' unless in_transaction?
+
+      super(table_name, new_name) # Actually rename the table
+      return if skip_view
```

Comment:
I'm quite unhappy with such a conditional approach.
Unfortunately, however, [`up_only`](https://api.rubyonrails.org/classes/ActiveRecord/Migration.html#method-i-up_only) is not available in this context.

Reply:
Sadly enough, due to some logic I don't fully understand [here](https://github.com/rails/rails/blob/2432988f9fc0a87fe7a8919ccd5cebd08d87593c/activerecord/lib/active_record/migration.rb#L731), neither [`up_only`](https://github.com/rails/rails/blob/2432988f9fc0a87fe7a8919ccd5cebd08d87593c/activerecord/lib/active_record/migration.rb#L807), [`reverting?`](https://github.com/rails/rails/blob/2432988f9fc0a87fe7a8919ccd5cebd08d87593c/activerecord/lib/active_record/migration.rb#L748) or a fancy [`reversible`](https://github.com/rails/rails/blob/2432988f9fc0a87fe7a8919ccd5cebd08d87593c/activerecord/lib/active_record/migration.rb#L788)
```ruby
reversible do |dir|
  dir.down { ... }
  dir.up { ... }
end
```
work **inside** migration methods.

I suppose those are purposed to run on the migration level.

I'll try to hack around `CommandRecorder`, as the only usable thing it does for `rename_table` is to swap the old and the new table names.